cmd/internal/provider/default_systems.go

```diff
@@ -0,0 +1,30 @@
+package provider
+
+import (
+	"slices"
+
+	"github.com/google/trillian/quota"
+	"github.com/google/trillian/storage"
+)
+
+var (
+	DefaultQuotaSystem   string
+	DefaultStorageSystem string
+)
+
+func init() {
+	defaultProvider := "mysql"
+	providers := storage.Providers()
+	if len(providers) > 0 && !slices.Contains(providers, defaultProvider) {
+		slices.Sort(providers)
+		defaultProvider = providers[0]
+	}
+	DefaultStorageSystem = defaultProvider
+
+	providers = quota.Providers()
+	if len(providers) > 0 && !slices.Contains(providers, defaultProvider) {
+		slices.Sort(providers)
+		defaultProvider = providers[0]
+	}
+	DefaultQuotaSystem = defaultProvider
```

Comment:
This may change the backward compatibility of using `mysql` as the default provider in the following assumption. I'm not sure if we want to update the behavior.

**Input**
storage.Providers(): `postgresql`
quota.Providers(): `mysql`, `postgresql`

**Want**
DefaultStorageSystem: `postgresql`
DefaultQuotaSystem: `mysql`

**Got**
DefaultStorageSystem: `postgresql`
DefaultQuotaSystem: `postgresql`

Reply:
This scenario shouldn't occur.  The build tags, if used, keep the quota and storage provider lists synchronised.  So if `mysql` is returned by `quota.Providers()` then it will also be returned by `storage.Providers()`, and vice versa.

Reply:
Great, thanks!